src/_ert/forward_model_runner/reporting/event.py

```diff
@@ -78,53 +75,46 @@ def __init__(self, evaluator_url, token=None, cert_path=None):
         self._real_id = None
         self._event_queue: queue.Queue[events.Event | EventSentinel] = queue.Queue()
         self._event_publisher_thread = ErtThread(target=self._event_publisher)
-        self._timeout_timestamp = None
-        self._timestamp_lock = threading.Lock()
-        # seconds to timeout the reporter the thread after Finish() was received
-        self._reporter_timeout = 60
+        self._done = threading.Event()
+        self._ack_timeout = ack_timeout
+        self._max_retries = max_retries
 
-    def stop(self) -> None:
+    def stop(self):
         self._event_queue.put(Event._sentinel)
-        with self._timestamp_lock:
-            self._timeout_timestamp = datetime.now() + timedelta(
-                seconds=self._reporter_timeout
-            )
+        self._done.set()
         if self._event_publisher_thread.is_alive():
             self._event_publisher_thread.join()
 
     def _event_publisher(self):
-        logger.debug("Publishing event.")
-        with Client(
-            url=self._evaluator_url,
-            token=self._token,
-            cert=self._cert,
-        ) as client:
-            event = None
-            while True:
-                with self._timestamp_lock:
-                    if (
-                        self._timeout_timestamp is not None
-                        and datetime.now() > self._timeout_timestamp
-                    ):
-                        self._timeout_timestamp = None
-                        break
-                if event is None:
-                    # if we successfully sent the event we can proceed
-                    # to next one
-                    event = self._event_queue.get()
-                    if event is self._sentinel:
-                        break
-                try:
-                    client.send(event_to_json(event))
-                    event = None
-                except ClientConnectionError as exception:
-                    # Possible intermittent failure, we retry sending the event
-                    logger.error(str(exception))
-                except ClientConnectionClosedOK as exception:
-                    # The receiving end has closed the connection, we stop
-                    # sending events
-                    logger.debug(str(exception))
-                    break
+        async def publisher():
+            async with Client(
+                url=self._evaluator_url,
+                token=self._token,
+                ack_timeout=self._ack_timeout,
+            ) as client:
+                event = None
+                start_time = None
+                while True:
+                    try:
+                        if self._done.is_set() and start_time is None:
+                            start_time = time.time()
+                        if event is None:
+                            event = self._event_queue.get()
```

Comment:
What would happen if we changed this from queue.Queue to asyncio.Queue? Would it fail due to no running event loop outside of the ErtThread?

Reply:
I will look if it can be refactored, but fear that it wouldn't work and reporter calls reporter.report from synced context.